docs/decisions/0026-file-service.md

```diff
@@ -0,0 +1,148 @@
+---
+# These are optional elements. Feel free to remove any of them.
+status: proposed
+contact: crickman, mabolan, semenshi
+date: 2024-01-16
+---
+
+# File Services
+
+## Context and Problem Statement
+OpenAI provides a file service for uploading files to be used for *assistant retrieval* or *model fine-tuning*: `https://api.openai.com/v1/files`
+
+Other providers may also offer some type of file-service, such as Gemini.
+
+> Note: *Azure Open AI* does not currently support the OpenAI file service API.
+
+## Considered Options
+
+1. Add OpenAI file service support to `Microsoft.SemanticKernel.Experimental.Agents`
+2. Add a file service abstraction and implement support for OpenAI
+3. Add OpenAI file service support without abstraction
+
+## Decision Outcome
+
+> Option 2. **Add a file service abstraction and implement support for OpenAI**
+
+Defining a generalized file service interface provides an extensibility point for other vendors, in addition to *OpenAI*.
+
+## Pros and Cons of the Options
+
+### Option 1. Add OpenAI file service support to `Microsoft.SemanticKernel.Experimental.Agents`
+**Pro:**
+1. No impact to existing AI connectors.
+
+**Con:**
+1. No reuse via AI connectors.
+1. No common abstraction.
+1. Unnatural dependency binding for uses other than with OpenAI assistants.
+
+### Option 2. Add a file service abstraction and implement support for OpenAI
+**Pro:**
+1. Defines a common interface for file service interactions.
+1. Allows for specialization for vendor specific services.
+
+**Con:**
+1. Other systems may diverge from existing assumptions.
+
+
+### Option 3. Add OpenAI file service support without abstraction
+**Pro:**
+1. Provides support for OpenAI file-service.
+
+**Con:**
+1. File service offerings from other vendors supported case-by-case without commonality.
+
+
+## More Information
+
+Signatures for *Option 2*, but can adapt to any option.
+
+#### `Microsoft.SemanticKernel.Abstractions`
+```csharp
+namespace Microsoft.SemanticKernel.Files;
+
+public interface IFileService : IAIService
```

Comment:
Should this be an abstract class to protect against breaking changes in the future if/when we need to add additional functionality.

Reply:
Doesn't a base class and interface respond the same for equivalent changes w.r.t. backward compat?

Reply:
Updated